---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Grab token using OAuth
      ansible.builtin.uri:
        url: "{{ SN_HOST }}/oauth_token.do"
        method: POST
        body_format: form-urlencoded
        return_content: true
        body:
          grant_type : password
          client_id : "{{ SN_CLIENT_ID }}"
          client_secret : "{{ SN_CLIENT_SECERT }}"
          username : "{{ SN_USERNAME }}"
          password : "{{ SN_PASSWORD }}"
      register: oauth_return_token

    - name: Ansible Reassign the CTASK if failed
      ansible.builtin.uri:
        url: "{{ SN_HOST }}/api/abb2/v2/generic_change/updateChangeTask"
        method: PATCH
        headers:
          Content-Type : application/json
          Accept : application/json
          Authorization : "Bearer {{ oauth_return_token.json.access_token }}"
        body_format: json
        body: 
          assignment_group: "{{ command_center_support_group }}"
          work_notes: "Error has been encountered during the automation run. Error Detail: {{ Science_Logic_message }}. Deploy the change manually "
          assigned_to : ""
          sys_id: "{{ change_task_sys_id }}"
      register: SR_reassign_ctask  
